hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/HoodieTableSource.java

```diff
@@ -365,16 +368,16 @@ private PartitionPruners.PartitionPruner createPartitionPruner(List<ResolvedExpr
         .build();
   }
 
-  private int getDataBucket(List<ResolvedExpression> dataFilters) {
+  private Option<Function<Integer, Integer>> getDataBucketFunc(List<ResolvedExpression> dataFilters) {
     if (!OptionsResolver.isBucketIndexType(conf) || dataFilters.isEmpty()) {
-      return PrimaryKeyPruners.BUCKET_ID_NO_PRUNING;
+      return Option.empty();
```

Comment:
> It looks like the return value is always non-empty, can we just return the function instead of the option instead?

Here may return `Option.empty()` when `getDataBucketFunc`